src/hooks/async/nft.ts

```diff
@@ -17,7 +17,7 @@ import invariant from "tiny-invariant";
  @internal
  */
 export function detectErc721Instance(
-  contract?: ValidContractInstance | SmartContract | null,
+  contract: ValidContractInstance | SmartContract | null | undefined,
```

Comment:
this is a good change, we should extract this into a generic type of
```typescript
type PossibleContract<{actual contract}>
```

Reply:
This will belong in the SDK right? If so i can make another PR for this